src/installer/corehost/cli/bundle/extractor.cpp

```diff
@@ -204,7 +204,7 @@ void extractor_t::extract_new(reader_t& reader)
     begin();
     for (const file_entry_t& entry : m_manifest.files) 
     {
-        if (entry.needs_extraction())
+        if (m_manifest.is_netcoreapp3_compat_mode() || entry.needs_extraction())
```

Comment:
It looks like there are a number of other places in this file that are wrong (`locate`, et al). I think a safer approach than trying to modify all the use sites is to have `file_entry_t::read` take an extra `force_extract` parameter, then pass it through to each `file_entry_t` constructor and store it in a field. Then we can check the field in `needs_extraction` and avoid changing all the call sites

Reply:
Agreed - the problem is everywhere we call `entry.needs_extraction`, so it would be easier to bake it into the `needs_extraction` itself. It would feel cleaner if we stored something like `extracted` on each file entry.